submissions/natashafir/memory-pair-game/index.js

```diff
@@ -0,0 +1,123 @@
+const items = [
+    {
+        id: 0,
+        img_url: './img/colibri.gif'
+    },
+    {
+        id: 1,
+        img_url: './img/hello.gif'
+    },
+    {
+        id: 2,
+        img_url: './img/hi.gif'
+    },
+    {
+        id: 3,
+        img_url: './img/inlove.gif'
+    },
+    {
+        id: 4,
+        img_url: './img/love.gif'
+    },
+    {
+        id: 5,
+        img_url: './img/play.gif'
+    },
+    {
+        id: 6,
+        img_url: './img/sleep.gif'
+    },
+    {
+        id: 7,
+        img_url: './img/space.gif'
+    }
+];
+
+let choosedCards = [];
+const sectionMemory = document.getElementById('section__memory');
+let hasFlippedCard = false;
+let boardLocked = false;
+let firstCard, secondCard;
+
+function shuffle(array) {
+    for (let i = array.length - 1; i > 0; i--) {
+        let j = Math.floor(Math.random() * (i + 1));
+        [array[i], array[j]] = [array[j], array[i]];
+    }
+    return array;
+}
+
+function createCardItem(item) {
+    const cardWrapper = document.createElement('div');
+    const imgBack = document.createElement('img');
+    const imgFront = document.createElement('img');
+    cardWrapper.classList.add("memory-card"); //class for CSS
+    cardWrapper.appendChild(imgBack);
+    cardWrapper.appendChild(imgFront);
+    imgBack.src = "./img/img-0.jpg";
+    imgFront.src = item.img_url;
+    imgBack.classList.add("back-face");
+    imgFront.classList.add("front-face");
+    cardWrapper.id = item.id;
+    return cardWrapper;
+}
+
+function createAllCards() {
+    let allCards = shuffle([...items, ...items]);
+    allCards.forEach(elem => {
+        sectionMemory.appendChild(createCardItem(elem));
```

Comment:
Adding elements to DOM from a loop is a bad practice. A browser will run reflow and repaint for every element in the loop. Instead, you can: 1. Use append method, which can add several elements in one operation 2. Create some wrapper, add your items to the wrapper and then add it to DOM. It will be one operation. 3. Clone current container. Add items to a container and then replace your old container with a new one. But be aware of event listeners. 4. Use innerHTML instead